src/gluonts/model/deepar/_estimator.py

```diff
@@ -186,9 +186,16 @@ def create_transformation(self) -> Transformation:
                     pred_length=self.prediction_length,
                     log_scale=True,
                 ),
+                SetFieldIfNotPresent(
+                    field=FieldName.FEAT_DYNAMIC_REAL, value=None
+                ),
```

Comment:
I am not sure I am convinced that this is the best way do implement this.

What will happen if some lines have `feat_dynamic_real` and some not? If I read this correctly, the output feature `time_feat` will have a variable dimension. 

Another issue is performance. In the current implementation, we will add a field `feat_static_real` with a default value `None` only to ignore it later in the `VstackFeatures` implementation.

I think a better way to do this is to have an `__init__` parameter in the estimator that indicates whether `feat_dynamic_real` is expected and should be used. Based on this flag we can do multiple things, e.g.

- infer a schema of a particular estimator instance, 
- throw an error if `feat_dynamic_real` is missing or has malformed input
- statically decide whether the `input_fields` argument of the `VstackFeatures` transformation below should contain `FieldName.FEAT_DYNAMIC_REAL` at the time when we construct the transformation pipeline.

Reply:
Yeah, that's the other way of doing it. I was trying to avoid those additional hyperparameters, but it's true that they make everything very explicit.

Reply:
@aalexandrov, this is already the behavior for FEAT_STATIC_CAT if I am not mistaken (see line 200). Perhaps we could work on letting the user specify a schema in a separate PR?

Reply:
> What will happen if some lines have feat_dynamic_real and some not? If I read this correctly, the output feature time_feat will have a variable dimension.

Maybe we can add this case to the tests?

Reply:
Wouldn't it be nicer to have the estimator pick the features up automatically if they are present? I can imagine its easy to forget to switch this on.

Reply:
@vafl we discussed the two options today. Roughly speaking, either:
1. The estimator, by default, uses *all* available features in the data, unless they are explicitly disabled.
2. The estimator, by default, does *not* use any of the features, regardless of the fact that they are present or not in the dataset, and to use features one must explicitly enable it.

(The option to enable/disable features is potentially useful for experimentation purposes, since this way one does not need to e.g. reformat the data in order to exclude some features.)

Option 1 was implemented in the [previous commit](https://github.com/awslabs/gluon-ts/pull/125/files/2a55317b6c1d52ce0c7b846acbd4d2362896e9ad), in case you want to check it out.

Potential drawbacks that I see in option 1:
* All seems automatic, but some things like `cardinality` aren't: the user may get the false impression that he's doing the right thing but if he doesn't set `cardinality` he's probably doing just as good as if the categorical features were ignored. Unless we go for a fully automatic mode (one pass through the data, validating the schema, understanding which fields are/aren't there, checking min/max of categorical features and so on), in which case everything is beautiful.
* If the first entry in the dataset has a some features that the second doesn't have, these will probably be transformed somewhat differently, and a weird and unreadable numpy or mxnet error will be thrown.

On the other hand, with option 2:
* You have to enable all optional features (i.e., fields which are not `"start"` or `"target"`) and setting the right `cardinality` is part of that.
* It's clear from the start what fields the transformation chain will expect to find: asserting that each data entry has those should be quite easy and yield a very clear error message in case the user is missing something.

I'm happy to discuss this further, since it's a delicate matter and a design choice that we should carry over to all other built-in estimators.